### PR TITLE
Updates z6.dmm, Derelict/DJ Station/Teleporter Remap & Clown Shuttle Fix

### PR DESCRIPTION
**Pull request by Shadeykins**
**What does this PR do:**
This PR makes a number of changes to Z6.

Notably it:
* Remaps the derelict entirely with a new donut-shaped KC13.
* Remaps the derelict teleporter for aesthetics so it's no longer a blank metal chunk in space.
* Remaps DJ Station to logically tie it in with KC13.
* Fixes the two absent tiles on the clown/mime shuttle crash that have irked me forever.

**Why did you do this?**
The derelict admittedly had next to no exploration element to it. While it served as a cool reference to the original map (something I only discovered last week admittedly), it did not provide players with anything truly meaningful in terms of worldspace--it was pretty much a drive-by loot fest or cult base.

The remap is predicated on the community's repeated wish for more lore/integration of the worldspace beyond what's simply physically there (as requested in the survey) and a desire to provide players with something interesting to see, to play with, to utilize, and to discover (some of the elements should even provide a laugh or two!).

I also wished to address the fact that many people express an interest in repairing the derelict, but never have the time to (it takes well over two hours to power/atmos it). The way in which the station is damaged and the tools it provides should make using it as a fun hangout much more viable.

By breaking up the map into discrete areas that are locked down, the derelict now demands some investigation in order to proceed/derive anything truly valuable out of it. You can't simply fly past it for free equipment anymore.

I will now go through the changes one by one, starting with the least changes to most.

**Clown/Mime Shuttle**
There was an adjacent wall on the clown shuttle missing which resulted in the asteroid turf (rather than rock) displaying under the half-walls on the shuttle. This has been fixed by manually putting in some asteroid mineral rocks (the regular rock spawn does not work in this case).

**Derelict Teleporter**
The derelict teleporter was more or less a chunk of metal with a table and a teleporter. It has been visually overhauled (in tiling/marginally in shape) to have visual context. It now plausibly looks like it was once part of something.

**DJ Station**
DJ Station was changed to be congruent with the new derelict, it also received some downgrading/upgrading.
* Adds free cigarette vendor w/ custom slogans
* Adds BODA machine.
* Removes disposal chute in favour of primitive bin
* Adds shower
* Lobby area made one tile wider. Telecomms are one tile shorter.
* Fake solar panels have been given wiring (they are still fake) so they're not weirdly disconnected.

**KC 13/Derelict**
The derelict was completely overhauled from the ground up. Here are the important object changes that have sincere/potential balance implications.
* Removes suspicious toolbox (x2), replaced with tool locker + combat gloves (x2)
* Adds prototype laser pistol (9 shots, or 1 taser shot).
* Adds double-barrel shotgun.
* Adds pepperspray.
* Adds minor and rudimentary departments (incredibly basic chemistry, kitchen).
* Adds some hostile mobs (space carp, Experiment 432-A) and an atmos hazard (bridge walkway).
* Everything else from the derelict has otherwise been maintained (white ship board, telebaton, classic EMP nade, three EVA suits, god emperor disk, SWAT helmet, etc.)
* Maintenance loot spawners to provide some level of randomization (10 in total).

Here are some of the unique features to this station.
* Security, Bridge, and Central Annex start locked down. Players will need to explore to lift them/progress through the derelict.
* Station has functional solar arrays/a functional SMES. If the station remains undiscovered for too long, explorers will need to repower it to proceed.
* Adds several minor lore elements such as items and cosmetics with custom names/descriptions. Several items which are not fluff but simply not in use (retro jumpsuits, hats, etc.) were placed to give the area unique flavour.
* Adds a backstory to the derelict beyond "we blew it up boys." There are things ranging from context clues on how to lift the lockdowns, to anecdotal things about the worldspace, to personal/amusing correspondence between Russian spacemen (there's even some bad poetry).
* Special variant of Officer Buzzsky called Officer Blyatsky. He wields the power of communism.
* It's totally circular, man.

**Images of sprite/map changes (IF APPLICABLE):**

I would like to give a special thanks to Kluys, FalseIncarnate, and Ansari for looking at it/making some suggestions while I was doing the design. I would like to give double-kudos to FreeStylaLT for helping me figure out Dreamdestroyer and to Fethas for helping me figure out PR majigs.

**Changelog:**
:cl:
add: New derelict station.
del: Old derelict station.
tweak: Remap teleporter, DJ station, adds vodka
fix: Clown shuttle missing adjacent wall tiles
/:cl:

### CODE REVIEW

**Comment by datlo** [comment]
Just FYI, the teleporter on z6 is not the abandonned teleporter but the derelict teleporter. The abandonned teleporter is next to the telecomms sat.

I'm not sure how I feel about this derelict remap. I need to look at it more and think it over.

**Comment by Shadeykins** [comment]
> 
> 
> Just FYI, the teleporter on z6 is not the abandonned teleporter but the derelict teleporter. The abandonned teleporter is next to the telecomms sat.
> 
> I'm not sure how I feel about this derelict remap. I need to look at it more and think it over.

good catch, updated comment to reflect this

**Comment by datlo** [comment]
As we discussed on discord, the walls need to be mineral titanium walls and not shuttle walls. 

You can reinforce them by making a subtype that's invulnerable to welding/thermite, but they need to be destroyable by at least explosions or otherwise cult bases could be super annoying with this.

I think I support this remap once the walls have been changed.

**Comment by Shadeykins** [comment]
> I think I support this remap once the walls have been changed.

addressed/fixed. There are some very rare edge cases where this is not possible (such as jagged edges pointing into space or half-tile blending) in order to produce the wanted effect. All other tiles are now destructible via explosion (about 99% of them).

**Comment by Quantum-M** [comment]
I think the plants in the derelict should be dead, considering that the derelict appears to had been abandoned a long time ago.

**Comment by Shadeykins** [comment]
> 
> 
> I think the plants in the derelict should be dead, considering that the derelict appears to had been abandoned a long time ago.

There is sadly only one dead plant type and it looks like wet spaghetti. You can see it in the derelict teleporter screenshot. If there were more dead plants I would have definitely used them.

**Comment by datlo** [comment]
Could just be cheap fake plants. Honestly; the ones on station probably are too.

**Comment by Fox-McCloud** [comment]
Making things shuttle walls isn't ideal; they have a lot of snowflakey unique behaviors that really shouldn't be utilized anywhere other than...well, shuttles.

**Comment by Shadeykins** [comment]
IconDiffbot hates me. RIP.

File: code/game/objects/items/trash.dm [hunk 1]
```diff
@@ -1,97 +1,103 @@
...
+
+/obj/item/trash/gum
+	name = "chewed gum"
+	desc = "NOT free candy."
+	icon_state = "gum"
+
+/obj/item/trash/tastybread
+	name = "bread tube"
+	icon_state = "tastybread"
+
```
Comment (Shadeykins) [inline code/game/objects/items/trash.dm]: new changes start here

File: code/game/objects/items/trash.dm [hunk 2]
```diff
@@ -1,97 +1,103 @@
...
+/obj/item/trash/spentcasing
+	icon = 'icons/obj/ammo.dmi'
+	name = "bullet casing"
+	desc = "A spent bullet casing. Smells like cordite."
+	icon_state = "gshell"
+
```
Comment (Shadeykins) [inline code/game/objects/items/trash.dm]: and stop here

File: code/game/turfs/simulated/walls_mineral.dm [hunk 3]
```diff
@@ -1,268 +1,293 @@
...
+	icon_state = "smooth"
+	smooth = SMOOTH_MORE|SMOOTH_DIAGONAL
+	canSmoothWith = list(/turf/simulated/wall/mineral/titanium/survival, /obj/machinery/door/airlock, /obj/structure/window/full, /obj/structure/window/full/reinforced, /obj/structure/window/full/reinforced/tinted, /obj/structure/window/full/shuttle, /obj/structure/shuttle/engine)
+
+/turf/simulated/wall/mineral/titanium/survival/nodiagonal
+	smooth = SMOOTH_MORE
+
+/turf/simulated/wall/mineral/titanium/survival/pod
+	canSmoothWith = list(/turf/simulated/wall/mineral/titanium/survival, /obj/machinery/door/airlock/survival_pod)
+
```
Comment (Shadeykins) [inline code/game/turfs/simulated/walls_mineral.dm]: changes start here

File: code/game/turfs/simulated/walls_mineral.dm [hunk 4]
```diff
@@ -1,268 +1,293 @@
...
+	icon = 'icons/turf/shuttle.dmi'
+	icon_state = "wall"
+
+//properties for derelict sub-type to prevent said deconstruction/thermiting
+/turf/simulated/wall/mineral/titanium/nodecon/try_decon(obj/item/I, mob/user, params)
+	return
+
+/turf/simulated/wall/mineral/titanium/nodecon/thermitemelt(mob/user as mob, speed)
+	return
+
```
Comment (Shadeykins) [inline code/game/turfs/simulated/walls_mineral.dm]: and stop here

File: code/game/turfs/simulated/walls_mineral.dm [hunk 5]
```diff
@@ -1,268 +1,293 @@
-/turf/simulated/wall/mineral
```
Comment (Shadeykins) [inline code/game/turfs/simulated/walls_mineral.dm]: Actual changes are from line 223-248. I did a bad and improperly deleted the file prior to re-submission.

**Comment by Shadeykins** [comment]
Closing to unfuck the changes listing.

**Pull request closed without merging**